src/back/glsl/mod.rs

```diff
@@ -353,6 +360,35 @@ impl<'a, W: Write> Writer<'a, W> {
         let mut namer = proc::Namer::default();
         namer.reset(module, keywords::RESERVED_KEYWORDS, &["gl_"], &mut names);
 
+        // Collect the list of unused, unsized structs for this entrypoint
+        let mut unused_unsized_types = Vec::default();
+        for (handle, global) in module.global_variables.iter() {
+            if info.get_entry_point(ep_idx)[handle].is_empty() {
+                match module.types[global.ty].inner {
+                    // If the global type is an unsized array
+                    crate::TypeInner::Array { size, .. } => {
+                        if crate::ArraySize::Dynamic == size {
+                            unused_unsized_types.push(global.ty);
```

Comment:
I don't think we can do this. According to this logic, if one global is used, and another one is unused, and they have the same type, this type would still go into the `unused_unsized_types`, even though it's used by the other global

Reply:
Yeah, you're right. I think I keep overcomplicating this. I've pushed another try.

This one should make more sense. :crossed_fingers: